src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -63,29 +80,52 @@ public JwtTokenReader(TokenValidationParameters tokenValidationParameters, strin
         /// <param name="token">The JWT token.</param>
         public async Task<bool> IsValidTokenAsync(string token)
         {
+            bool result;
+
             try
             {
-                TokenValidationParameters validationParameters = await DetermineTokenValidationParametersAsync();
-                _handler.ValidateToken(token, validationParameters, out SecurityToken jwtToken);
+                var validationParameters = await DetermineTokenValidationParametersAsync();
+                this._handler.ValidateToken(token, validationParameters, out var validatedToken);
+                result = ValidateClaimCheck(validatedToken);
             }
-            catch (Exception exception)
+            catch
             {
                 return false;
             }
 
-            return true;
+            return result;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="validatedToken"></param>
+        /// <returns></returns>
+        private bool ValidateClaimCheck(SecurityToken validatedToken)
+        {
+            if (!this._claimCheck.Any()) 
+                return true;
+
+            var jwtToken = (JwtSecurityToken) validatedToken;
+
+            var validClaims = (from c in jwtToken.Claims
+                join m in this._claimCheck on c.Type equals m.Key
+                where c.Value == m.Value
+                select c).Count();
+
+            return this._claimCheck.Count == validClaims;
         }
 
         private async Task<TokenValidationParameters> DetermineTokenValidationParametersAsync()
         {
-            if (_tokenValidationParameters is null)
+            if (this._tokenValidationParameters is null)
             {
-                OpenIdConnectConfiguration config = await _configManager.GetConfigurationAsync();
-                
+                var config = await this._configManager.GetConfigurationAsync();
+
                 var validationParameters = new TokenValidationParameters
                 {
                     ValidateAudience = false,
-                    ValidAudience = _applicationId,
+                    ValidAudience = !this._claimCheck.Any() ? this._applicationId : string.Empty,
```

Comment:
You can check if there exists a ClaimCheck for the audience, and use that one instead of this construct.
Next to that, why is `ValidateAudience` set to false if `ValidAudience` has a value ?

(I see that this was already the case in the previous version as well ?  https://docs.microsoft.com/en-us/dotnet/api/microsoft.identitymodel.tokens.tokenvalidationparameters?view=azure-dotnet#properties)

Reply:
Actually, we could also choose to not set the  `ValidAudience` property, and rely on the claimcheck-check only ...

Reply:
done